presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/trait/CardinalityTraitType.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.sql.planner.iterative.trait;
+
+import com.facebook.presto.sql.planner.iterative.TraitType;
+
+public class CardinalityTraitType
```

Comment:
Lacks `toString()`. The fact this implements `equals()` and `hashCode` sufficiently is not quite obvious. Consider implementing this as an enum.

Reply:
There is going to be a single instance of this class, so I will remove `equals` and `hashCode`.

It used to be an enum. I changed this on @sopel39 suggestion. This way it is easy to add dynamically new traits.

Reply:
enums are not pluggable

Reply:
I say `CardinalityTraitType` could be an enum. I didn't mean `TraitType` to be one.

Reply:
eg
```
public enum CardinalityTraitType
     implements TraitType<CardinalityTrait> 
{
     CARDINALITY;

      @Override
      public boolean isEquivalenceGroupApplicable()
      {
          return true;
      }
}
```

Reply:
> There is going to be a single instance of this class, so I will remove equals and hashCode.

Maybe there should be `String TraitType::signature()` method which will be used to compare `TraitTypes`. Then we could:
* directly compare types using `signature`
* have some kind of `AbstractTraitType` that would implement `equals` which would use `signature` to compare types, see: `AbstractType#equals`

I prefer the former since it is more explicit.

Reply:
We still need proper contract for `equals` and `hashCode`.